src/framework/ui/qml/MuseScore/Ui/internal/ProgressDialog.qml

```diff
@@ -63,6 +63,8 @@ StyledDialogView {
 
         spacing: 16
 
+        onHeightChanged: root.updateGeometryOnHeightChange = true
```

Comment:
How is this supposed to work? Are we hoping that this is called first, and then the `onHeightChanged` handler in `StyledDialogView` is called? Is there any guarantee that the handlers will indeed be called in this order?

_If_ we do have that guarantee, that seems to imply that `updateGeometry()` will simply always be called in `onHeightChanged` for `ProgressDialog`, and never in other dialogs. 
If that's a correct analysis, then why would we make it so complicated? Why not just `onHeightChanged: root.updateGeometry()` in `ProgressDialog`, and leave `StyledDialogView` untouched? 
Also, this solution feels a bit incomplete; it does not solve the problem with other dialogs. Of course, that was not part of this issue, but still it would be nice if we can fully solve this problem.

Have you tried my suggestion from https://github.com/musescore/MuseScore/pull/18405#discussion_r1253064667? And what were the results of that?